src/Modules/TasksModule/components/TaskList/TaskList.tsx

```diff
@@ -1,5 +1,328 @@
-import React from "react";
+import axios from "axios";
+import { useCallback, useEffect, useState } from "react";
+import Modal from "react-bootstrap/Modal";
+import ResponsivePagination from "react-responsive-pagination";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import { TaskInterface } from "../../../../interfaces/Auth";
+import {
+  baseUrl,
+  handleApiError,
+  loader,
+  getRequestHeaders,
+} from "../../../../utils/Utils";
+import DeleteData from "../../../SharedModule/components/DeleteData/DeleteData";
+import NoData from "../../../SharedModule/components/NoData/NoData";
+import style from "../Tasks.module.css";
 
-export default function TaskList() {
-  return <>TaskList</>;
+export default function TasksList() {
+  const navigate = useNavigate();
+  const [tasks, setTasks] = useState<[TaskInterface] | []>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  // State for page number
+  const [pageNumber, setPageNumber] = useState(1);
```

Comment:
I don't know if I mentioned that here or not, but for pagination and filtration. It is not recommended to use a normal state for storing them. If we reload the page, everything will be lost. It is reccomended to use something like `useSearchParams` from react-router-dom. It will store the state in the URL and preserve it when reloading. 

You might leave everything as it is in this project, but in the next one , we need to consider using it.